js/runtime.js

```diff
@@ -1516,7 +1521,18 @@
             getYear: function(date){
                 return date.getFullYear();
             },
-
+            validateYear: function(inputElement){
```

Comment:
Alternatively we could parse the number and make sure it is in the range 1-9999 as an integer.